### PR TITLE
[Merged by Bors] - Update libp2p

### PR DESCRIPTION
**Pull request by AgeManning**
This is a little bit of a tip-of-the-iceberg PR. It houses a lot of code changes in the libp2p dependency. 

This needs a bit of thorough testing before merging. 

The primary code changes are:
- General libp2p dependency update
- Gossipsub refactor to shift compression into gossipsub providing performance improvements and improved API for handling compression

### CODE REVIEW

File: beacon_node/eth2_libp2p/src/types/pubsub.rs
Comment (paulhauner) [inline beacon_node/eth2_libp2p/src/types/pubsub.rs:84]: I would have thought we need to check the length of `data` instead of `compressed_data`.

**Review by paulhauner — CHANGES_REQUESTED** [review]
I've gone through your changes and it looks good accept for two things:

1. I left a comment about checking message size. I don't think it's something likely to cause an issue, but perhaps worth fixing anyway (if its indeed an error).
1. I added b9bd617cc, which is a trivial change that should reduce some allocations for each gossip message we ID. It's not a big deal, we can just drop it if you think it will cause issue.

Happy to merge once you've addressed (1) and approve of (2).

I haven't gone through the libp2p changes, since it's a [huge diff](https://github.com/sigp/rust-libp2p/compare/830e6fabb7ee51281a98f5e092f056668adbef25..97000533e4710183124abde017c6c3d68287c1ae) and I assume it's already had eyes across it from the libp2p devs.

File: beacon_node/eth2_libp2p/src/types/pubsub.rs
Comment (AgeManning) [reply beacon_node/eth2_libp2p/src/types/pubsub.rs]: Yeah, this is outbound, we shouldn't hit it, but would be stricter to put it on data, have updated

**Comment by AgeManning** [comment]
Changes in 1 seem fine

**Review by paulhauner — APPROVED** [review]
LGTM!

bors r+

**Pull request closed without merging**
